src/components/views/rooms/RoomHeader.js

```diff
@@ -159,6 +159,10 @@ module.exports = createReactClass({
             <E2EIcon status={this.props.e2eStatus} /> :
             undefined;
 
+        const joinRules = this.props.room && this.props.room.currentState.getStateEvents("m.room.join_rules", "");
+        const joinRule = joinRules && joinRules.getContent().join_rule;
+        const joinRuleClass = "mx_RoomHeader_PrivateIcon" + (joinRule === "invite" ? " mx_RoomHeader_isPrivate" : "");
```

Comment:
could this use the classnames module for readibility?

Reply:
oh, neat, I'd not seen that before but it's totally in the rest of hte file